src/java.base/share/classes/sun/security/rsa/RSACore.java

```diff
@@ -145,7 +152,7 @@ public static int getByteLength(RSAKey key) {
     public static byte[] rsa(byte[] msg, RSAPrivateKey key, boolean verify)
             throws BadPaddingException {
         if (key instanceof RSAPrivateCrtKey) {
-            if (useNativeRsa && key instanceof sun.security.rsa.RSAPrivateCrtKeyImpl) {
+            if (useNativeRsa && key instanceof sun.security.rsa.RSAPrivateCrtKeyImpl && (NativeCrypto.getVersion() == 0 && !unsupportedKeySizes.contains(key.getModulus().bitLength()))) {
```

Comment:
That comment just points back here, and I couldn't find the answer here either. Did I just miss the explanation?

Reply:
I should've pointed to the issue instead of the PR. I updated the comment to point to the issue and added a better summary of the issue and how its solved in [a comment[(https://github.com/eclipse-openj9/openj9/issues/11500#issuecomment-833004636)

Reply:
The new comment is better, but what is still not clear is why 1025 is problematic. Is it because it's not a power of 2 (so 1023, 2047 and 2049 would also be padded)? If so, the code should be corrected to properly identify when padding will occur; or better, be independent of openssl implementation details.